### PR TITLE
[Refactor] Disable pipeline load when need parallel

### PR DESCRIPTION
**Pull request by meegoo**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [x] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/planner/PlanFragment.java [hunk 1]
```diff
@@ -460,6 +460,14 @@ public void setSink(DataSink sink) {
         Preconditions.checkNotNull(sink);
         sink.setFragment(this);
         this.sink = sink;
+
+        // we only can do this here since finialize() had been removed before
+        if (ConnectContext.get() != null) {
+            if (!canUsePipeline()) {
+                this.parallelExecNum = ConnectContext.get().getSessionVariable().getParallelExecInstanceNum();
+                this.pipelineDop = 1;
```
Comment (chaoyli) [inline fe/fe-core/src/main/java/com/starrocks/planner/PlanFragment.java:468]: Why to set the pipelineDop = 1, but not `enable_pipeline_engine=false`?

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 16 / 16 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/DeletePlanner.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadingTaskPlanner.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/UpdatePlanner.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/PlanFragment.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/InsertPlanner.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by meegoo** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 9 / 9 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadingTaskPlanner.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/InsertPlanner.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by chaoyli — DISMISSED** [review]

**Comment by meegoo** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by meegoo** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 9 / 9 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadingTaskPlanner.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/InsertPlanner.java|2|2|100.00%|[]|

**Review by chaoyli — APPROVED** [review]

**Review by imay — APPROVED** [review]

**Comment by meegoo** [comment]
@backport branch-2.4

**Comment by meegoo** [comment]
@mergify backport branch-2.4

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 9 / 9 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadingTaskPlanner.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/InsertPlanner.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 9 / 9 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadingTaskPlanner.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/InsertPlanner.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 9 / 9 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadingTaskPlanner.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/InsertPlanner.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 9 / 9 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadingTaskPlanner.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/InsertPlanner.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 9 / 9 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadingTaskPlanner.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/InsertPlanner.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 9 / 9 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadingTaskPlanner.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/InsertPlanner.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 9 / 9 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadingTaskPlanner.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/InsertPlanner.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 9 / 9 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadingTaskPlanner.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/InsertPlanner.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 9 / 9 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadingTaskPlanner.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/InsertPlanner.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 9 / 9 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadingTaskPlanner.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/InsertPlanner.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 9 / 9 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadingTaskPlanner.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/InsertPlanner.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 9 / 9 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadingTaskPlanner.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/InsertPlanner.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 9 / 9 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadingTaskPlanner.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/InsertPlanner.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 9 / 9 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadingTaskPlanner.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/InsertPlanner.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 9 / 9 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadingTaskPlanner.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/InsertPlanner.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 9 / 9 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadingTaskPlanner.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/InsertPlanner.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 9 / 9 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadingTaskPlanner.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/InsertPlanner.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 9 / 9 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadingTaskPlanner.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/InsertPlanner.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 9 / 9 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadingTaskPlanner.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/InsertPlanner.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 9 / 9 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadingTaskPlanner.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/InsertPlanner.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 9 / 9 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadingTaskPlanner.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/InsertPlanner.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 9 / 9 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadingTaskPlanner.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/InsertPlanner.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 9 / 9 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadingTaskPlanner.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/InsertPlanner.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 9 / 9 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadingTaskPlanner.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/InsertPlanner.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 9 / 9 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadingTaskPlanner.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/InsertPlanner.java|2|2|100.00%|[]|

**Pull request merged**